### ISSUE TITLE
Replace alternate repository URLs with standard ones

### ISSUE THREAD

**Issue by ericis** [orchidhq/Orchid#220]
Corporate environments are generally constrained to standard Java repositories. It's not even an option for our team.

Consider replacing the following repos with standard support and a single dependency required to run Orchid:
```
        jcenter()
        maven { url "https://kotlin.bintray.com/kotlinx" }
        maven { url 'https://dl.bintray.com/javaeden/Orchid/' }
        maven { url 'https://dl.bintray.com/javaeden/Eden/' }
        maven { url 'https://jitpack.io' }
```

**Comment by cjbrooks12** [comment]
This should definitely be do-able, I've just been lazy. Most of those extra repositories are just for dependencies of plugins or due to sync issues with Bintray (I'll list which repositories are needed by which plugins at the end of this post), and if you don't use that plugin then you don't need that repository. It also looks like most of Orchid's dependencies are available on Maven Central (except for those that I manage). 

What are considered the "standard Java repositories" that your team is able to use? Maven Central and JCenter? Just Maven Central? And which plugins are you interested in using? This will help me prioritize the work of moving those artifacts to the appropriate repository.

**Plugin->Repository List**

| Plugin                  | Artifact Type  | Repositories                             |
| ----------------------- | -------------- | ---------------------------------------- |
| OrchidCore              | core framework |                                          |
| OrchidAsciidoc          | plugin         |                                          |
| OrchidBible             | plugin         | jitpack                                  |
| OrchidDiagrams          | plugin         |                                          |
| OrchidSyntaxHighlighter | plugin         |                                          |
| OrchidWritersBlocks     | plugin         |                                          |
| OrchidChangelog         | plugin         |                                          |
| OrchidForms             | plugin         | Orchid bintray                           |
| OrchidJavadoc           | plugin         | Orchid bintray, jitpack, kotlinx bintray |
| OrchidKotlindoc         | plugin         | Orchid bintray, jitpack, kotlinx bintray |
| OrchidKSS               | plugin         |                                          |
| OrchidNetlifyCMS        | plugin         |                                          |
| OrchidPages             | plugin         | Orchid bintray                           |
| OrchidPluginDocs        | plugin         |                                          |
| OrchidPosts             | plugin         | Orchid bintray                           |
| OrchidPresentations     | plugin         | Orchid bintray                           |
| OrchidSearch            | plugin         |                                          |
| OrchidSwagger           | plugin         |                                          |
| OrchidSwiftdoc          | plugin         | Orchid bintray                           |
| OrchidTaxonomies        | plugin         |                                          |
| OrchidWiki              | plugin         |                                          |
| OrchidBsDoc             | theme          | Orchid bintray                           |
| OrchidEditorial         | theme          | Orchid bintray                           |
| OrchidFutureImperfect   | theme          |                                          |
| OrchidAll               | bundle         | Orchid bintray, jitpack, kotlinx bintray |
| OrchidBlog              | bundle         | jitpack                                  |
| OrchidLanguagePack      | bundle         | jitpack, kotlinx bintray                 |

**Comment by ericis** [comment]
Thanks for the quick and thorough response @cjbrooks12 

I will need to run a simple test against `jcenter()` on the corporate network and will report back.

We had two ideas:
1. replace `JavaDoc` for Java libraries with Orchid and integrate them into a `gh-pages` on our internal GitHub Enterprise instance (the `gh-pages` site is running an Angular website where we currently copy the doc files generated from JavaDoc and link to them with an aweful user transition between the site and the docs)
2. We REALLY want to document our Jenkins Shared Library written in Groovy. We were hopeful Orchid might "just work", since we're generating JavaDoc output already with `./gradlew groovydoc`. But, we were concerned since we saw Orchid Java and Kotlin as explicit support; assuming that maybe Groovy would need a new plug-in.

**Comment by cjbrooks12** [comment]
I just thought of a way to get around the multiple repositories now, without any changes to Orchid or its dependencies. 

You can set up Orchid using the kscript method [outlined in the docs](https://orchid.netlify.com/wiki/user-manual/getting-started/quickstart#kscript) rather than using Gradle. Kscript has an option to compile the script as a [standalone binary](https://github.com/holgerbrandl/kscript#deploy-scripts-as-standalone-binaries), which embeds the script and all of its dependencies. Basically makes a fat jar out of it, so running that binary does not need to do any dependency resolution. Just check in that binary and you're secure. 

This is how I build [my personal website](https://github.com/cjbrooks12/caseyjbrooks), so that I don't need to install kscript on Netlify. You can check out how I do it there for reference.

**Comment by ericis** [comment]
Wow! As I learn more about the "product suite" of Orchid, it's a fairly rich ecosystem of solutions. Thanks for the tip on the binary. It's worth an experiment to see how far we can get.

Any concerns about support for GroovyDoc currently generating JavaDocs?

**Comment by cjbrooks12** [comment]
1) This is the exact reason why I started developing Orchid in the first place. Trying to host a website AND your javadocs with it is just painful and a horrible user experience. Orchid's publication pipeline includes support for deploying to Github Pages, and it has an option to specify the enterprise URL. I don't actually use Github Enterprise at all so I can't test it, but I based it on what several other similar tools did so it should work.
2) Unfortunately, groovydoc is its own tool which won't have drop-in support, despite its similarty to Javadoc. It will need its own plugin. I haven't looked too hard to know if the Groovydoc tool has support for "doclets" like Javadoc does, which would give us a full documentation model, but at the very least Orchid could process the HTML output and either embed large chunks into the site, or try to scrape specific data from it. I personally don't use Groovy beyond Gradle, so I don't exactly know the special groovy-specific things that would be needed from it, but I would definitely be intersted in starting development on this plugin for you.

**Comment by ericis** [comment]
Added #221 

I'll do my best to run some experiments over the next few weeks and provide some feedback about what I experience behind a corporate firewall/proxy. Thanks for the details.

**Comment by cjbrooks12** [comment]
Update on this: 

The Kotlindoc and Javadoc plugins have been updated in 0.15.4 so they no longer do any runtime dependency resolution. Bundling your app as a packaged Kscript app should work completely offline. I just did some basic testing with my wifi turned off and it's working great.

I've also been doing work to get many of these dependencies hosted on JCenter, I've updated the table above to reflect these changes. In short:

- `https://dl.bintray.com/javaeden/Orchid/` and `https://dl.bintray.com/javaeden/Eden/` repositories are no longer needed at all
- Jitpack is only needed for the `OrchidBible` plugin and the bundles that include it
- `https://kotlin.bintray.com/kotlinx` is still needed for the Kotlindoc and Javadoc plugins and the bundles that include them
- All Orchid artifacts and other libraries I develop are on JCenter
